pkg/storage/stores/shipper/downloads/table_manager.go

```diff
@@ -185,12 +185,15 @@ func (tm *TableManager) syncTables(ctx context.Context) error {
 	tm.tablesMtx.RLock()
 	defer tm.tablesMtx.RUnlock()
 
+	start := time.Now()
 	var err error
 
 	defer func() {
 		status := statusSuccess
 		if err != nil {
 			status = statusFailure
+		} else {
+			tm.metrics.tablesSyncOperationDurationSeconds.Set(time.Since(start).Seconds())
```

Comment:
I'm curious why we only track this if there was no error? What if this runs for a long time and then fails?